tests/connection_test.ts

```diff
@@ -37,4 +37,58 @@ suite.test("swapdb", async () => {
   assertEquals(await client.swapdb(0, 1), "OK");
 });
 
+suite.test("parse basic URL", () => {
+  const options = parseURL("redis://127.0.0.1:7003");
+  assertEquals(options.hostname, "127.0.0.1");
+  assertEquals(options.port, 7003);
+  assertEquals(options.tls, false);
+  assertEquals(options.db, undefined);
+  assertEquals(options.name, undefined);
+  assertEquals(options.password, undefined);
+});
+
+suite.test("parse complex URL", () => {
+  const options = parseURL("rediss://username:password@127.0.0.1:7003/1");
+  assertEquals(options.hostname, "127.0.0.1");
+  assertEquals(options.port, 7003);
+  assertEquals(options.tls, true);
+  assertEquals(options.db, 1);
+  assertEquals(options.name, "username");
+  assertEquals(options.password, "password");
+});
+
+suite.test("parse URL with search options", () => {
+  const options = parseURL(
+    "redis://127.0.0.1:7003/?db=2&password=password&ssl=true",
+  );
+  assertEquals(options.hostname, "127.0.0.1");
+  assertEquals(options.port, 7003);
+  assertEquals(options.tls, true);
+  assertEquals(options.db, 2);
+  assertEquals(options.name, undefined);
+  assertEquals(options.password, "password");
+});
+
+suite.test("Check parameter parsing priority", () => {
+  const options = parseURL(
+    "rediss://username:password@127.0.0.1:7003/1?db=2&password=password2&ssl=false",
+  );
+  assertEquals(options.hostname, "127.0.0.1");
+  assertEquals(options.port, 7003);
+  assertEquals(options.tls, true);
+  assertEquals(options.db, 1);
+  assertEquals(options.name, "username");
+  assertEquals(options.password, "password");
+});
+
+suite.test("connect with URL", async () => {
```

Comment:
Personally, I think this test (`connect with URL`) is unnecessary :thinking: This is because tests for connecting to Redis have already been written :smiley:

Reply:
Just one more point: I think tests for parseURL should be in `general_test.ts` :smiley:

Reply:
Moved the tests to correct location and removed the connection test